### PR TITLE
fix: comparison ops

### LINKED ISSUE #244: Comparison with quantity not working
Example:
```python
x = Quantity(jnp.array([1., 0, -1.0]), u.kpc)
x > 0.0  # returns incorrect boolean array
x > 1.0  # should fail, instead returns incorrect boolean array
```

### PR DESCRIPTION
**Pull request by nstarman**
Fixes #244

### CODE REVIEW

**Comment by nstarman** [comment]
The `block_until_ready` error appears to be unrelated. But it would be good to solve. I'm not getting it on my machine.

**Comment by nstarman** [comment]
Apparently it is related #246 isn't triggering it.

**Comment by nstarman** [comment]
Found it. It's the new version of equinox. 0.11.8.

**Comment by nstarman** [comment]
@adrn should be ready to review

**Comment by nstarman** [comment]
I actually need to cut a release. This can be a bugfix release to follow up.

File: tests/unit/test_quantity.py [hunk 1]
```diff
@@ -233,7 +233,9 @@ def test_gt():
     assert jnp.array_equal(q > Quantity(1, u.m), jnp.array([False, True, True]))
 
     # Test with incompatible units
-    assert jnp.array_equal(q > Quantity(0, u.s), jnp.array([False, False, False]))
+    # TODO: better equinox exception matching
```
Comment (adrn) [inline tests/unit/test_quantity.py:236]: Here and below: Also add tests that `q > 0`, `q >= 0`, etc. should work even without units?

Comment (nstarman) [reply tests/unit/test_quantity.py:236]: SGTM.
Is this fix alight except for adding some tests? It changes exception types.

Comment (nstarman) [reply tests/unit/test_quantity.py:236]: @adrn, resolved?

**Pull request merged**